### PR TITLE
feat(login): auto-open browser for Auth0 and MFA authentication URLs

### LINKED ISSUE #121: Auto-open browser for MFA and login authentication URLs
## Summary

When the CLI requires browser-based authentication (MFA or initial login), it only prints the URL to stderr. Users must manually copy-paste the URL into their browser, which is cumbersome — especially when MFA is triggered repeatedly during a session.

**Current behavior:**
```
MFA authentication required. Please visit:
  https://auth.alpacon.io/authorize?response_type=code&client_id=...

⠋ Waiting for MFA authentication.
```

**Expected behavior:**
The CLI should automatically open the URL in the user's default browser, with a fallback to printing the URL if no browser is available.

## Affected Code

Two locations print auth URLs without opening a browser:

1. **MFA** — `api/mfa/mfa.go:40`
   ```go
   fmt.Fprintf(os.Stderr, "\nMFA authentication required. Please visit:\n  %s\n\n", mfaURL)
   ```

2. **Login (Auth0 device flow)** — `cmd/login.go:137`
   ```go
   fmt.Fprintf(os.Stderr, "\nPlease authenticate by visiting:\n  %s\n\n", utils.Blue(deviceCode.VerificationURIComplete))
   ```

## Suggested Implementation

Use Go's `exec.Command` to open the URL based on the platform:

```go
func openBrowser(url string) error {
    var cmd *exec.Cmd
    switch runtime.GOOS {
    case "darwin":
        cmd = exec.Command("open", url)
    case "linux":
        cmd = exec.Command("xdg-open", url)
    case "windows":
        cmd = exec.Command("rundll32", "url.dll,FileProtocolHandler", url)
    default:
        return fmt.Errorf("unsupported platform")
    }
    return cmd.Start()
}
```

Or use an existing library like [`pkg/browser`](https://github.com/pkg/browser) which handles all of this.

**Fallback strategy:**
- Attempt to open browser
- If it fails (e.g., headless server, SSH session), fall back to printing the URL as-is (current behavior)
- Optionally check for `SSH_CONNECTION` env var to skip the attempt on remote sessions

**Optional flag:**
- `--no-browser` flag to disable auto-open for users who prefer manual copy-paste or scripted environments

## Context

During a debugging session using `alpacon websh` across multiple servers, MFA re-authentication was required repeatedly. Each time, the URL had to be manually copied from the terminal and pasted into a browser. Automating this would significantly improve the CLI experience.

### PR DESCRIPTION
**Pull request by eunyoung14**
## Summary

Automatically open authentication URLs in the user's default browser during login (Auth0 device flow) and MFA prompts. The URL is always printed to stderr first as a fallback. Includes multiple safeguards for environments where a browser is unavailable, and a cross-process debounce to prevent multiple tabs when concurrent CLI sessions trigger MFA simultaneously.

## Changes

- **`utils/browser.go`** (new): `OpenBrowser()` with platform detection (darwin/linux/windows), SSH session skip (`SSH_CONNECTION`, `SSH_TTY`), Linux headless skip (`DISPLAY`/`WAYLAND_DISPLAY`), URL scheme validation, `ALPACON_NO_BROWSER` env var support, file-based debounce (`~/.alpacon/.browser_lock`, 30s window), zombie process prevention (`go cmd.Wait()`)
- **`utils/browser_test.go`** (new): 11 tests for `shouldOpenBrowser` (URL scheme, SSH, env var) and 3 tests for `acquireBrowserLock` (acquire, debounce block, expiry)
- **`api/mfa/mfa.go`**: Added `utils.OpenBrowser(mfaURL)` after MFA URL print
- **`cmd/login.go`**: Added `utils.OpenBrowser()` after Auth0 device flow URL print, `--no-browser` flag, updated Long/Example help text
- **`README.md`**: Documented `--no-browser` flag and `ALPACON_NO_BROWSER` env var in Login & Logout section

### Universal
- [x] Code follows project style guidelines
- [x] Self-reviewed the code
- [x] No hardcoded secrets or credentials
- [x] Updated relevant documentation

### Golang
- [x] Error handling is complete
- [x] No goroutine leaks (`go cmd.Wait()` reaps child process)
- [x] Tests include edge cases
- [x] `go vet` passes clean

Closes #121

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/login.go [hunk 1]
```diff
@@ -18,33 +18,39 @@ import (
...
 )
 
 var loginCmd = &cobra.Command{
 	Use:   "login",
 	Short: "Log in to Alpacon",
-	Long:  "Log in to Alpacon. To access Alpacon, workspace url is must specified",
-	Example: `
-	# Re-login to saved workspace
-	alpacon login
+	Long: `Log in to Alpacon. A workspace URL must be specified.
```
[the remark this answers is not in the document]
Comment (eunyoung14) [reply cmd/login.go]: Fixed. Updated to: "If no workspace URL is provided, the previously saved workspace is used, or you will be prompted to enter one."
